src/bp/ui-studio/src/web/views/FlowBuilder/diagram/index.tsx

```diff
@@ -152,12 +153,66 @@ class Diagram extends Component<Props> {
     }
   }
 
+  handleContextMenuNoElement = (event: React.MouseEvent) => {
+    const { x, y } = this.manager.getRealPosition(event)
+    const addFlowNode = () => {
+      this.props.createFlowNode({ x, y, type: 'standard' })
+    }
+    const addSkillNode = (id: string) => {
+      this.props.buildSkill({ location: { x, y }, id })
+    }
+    const addSayNode = () => {
+      this.props.createFlowNode({ x, y, type: 'say_something', next: [defaultTransition] })
+    }
+    const addExecuteNode = () => {
+      this.props.createFlowNode({ x, y, type: 'execute', next: [defaultTransition] })
+    }
+    const addListenNode = () => {
+      this.props.createFlowNode({ x, y, type: 'listen', onReceive: [], next: [defaultTransition] })
+    }
+    const addRouterNode = () => {
+      this.props.createFlowNode({ x, y, type: 'router' })
+    }
```

Comment:
What do you think of moving those methods to the main class, so we can use them in handleToolDropped ? This way we will only have to change them in one place in the future